### ISSUE TITLE
Global Update Feature for npm Dependencies in Turbo Repo

### ISSUE THREAD

**Issue by vaibhavmuchandi** [vercel/turbo#5669]
### Which project is this feature idea for?

Turborepo

### Describe the feature you'd like to request

Currently, in Turbo Repo, when an npm dependency package gets an update, developers must manually update it in each app or package using the `pnpm update` command. I propose a feature that allows for global updating of npm dependencies across all apps and packages. This would streamline the update process, saving time and reducing the risk of inconsistencies.

### Describe the solution you'd like

I would like to see the implementation of a command or configuration option within Turbo Repo that allows developers to update npm dependencies across all apps and packages simultaneously. This could be a new command such as `turbo update` or an extension of the existing update functionality with a global option.

### Describe alternatives you've considered

An alternative solution could be to create a script that automates the current process of individually updating dependencies in each app or package. However, integrating this feature directly into Turbo Repo would provide a more robust and user-friendly solution, aligning with the tool's goal of simplifying development workflows.

**Comment by vaibhavmuchandi** [comment]
For anyone else struggling with the similar issue as mine and wants to update npm dependencies in every app or package of their turbo repo can use [this](https://www.npmjs.com/package/turbo-repo-updater).

**Comment by NicholasLYang** [comment]
Hi @vaibhavmuchandi, thanks for the issue. We've been discussing some more integrations with package managers, so we can definitely add this as a potential feature.

**Comment by weyert** [comment]
Doesn't `syncpack` do something similar or am I  misunderstanding what you want to do?
https://jamiemason.github.io/syncpack/

**Comment by JamieMason** [comment]
There is `syncpack update` (https://jamiemason.github.io/syncpack/update). 

Examples

- Update react-related packages only: `syncpack update -f react`
- Update production dependencies only `syncpack update --types prod`

Or run with no options to update everything in the whole monorepo, although this would include peerDependencies, overrides/resolutions, etc

**Comment by weyert** [comment]
Yes, it's pretty awesome too! 
Lazy me sometimes wishes it would update versions or add overrides to resolve audit/security vulnerabilities :D

**Comment by JamieMason** [comment]
Thanks! There is [`npm audit --fix`](https://docs.npmjs.com/cli/v9/commands/npm-audit?v=true#examples) but are you looking for something a little different? That but monorepo-wide and/or changes your package.json instead of your lockfile?

**Comment by tknickman** [comment]
We definitely recommend `syncpack` (in fact we're going to have to use it on this repo soon enough) - but I'm sure there a few tools that solve the same issue out there!

**Comment by weyert** [comment]
> Thanks! There is [`npm audit --fix`](https://docs.npmjs.com/cli/v9/commands/npm-audit?v=true#examples) but are you looking for something a little different? That but monorepo-wide and/or changes your package.json instead of your lockfile?

Yeah, mainly to change in the `package.json` instead of lock file. That communicates better and avoids accidentally reverting fixes when not re-running `npm audit --fix`. Mainly just being lazy to be honest :)

**Comment by JamieMason** [comment]
> We definitely recommend `syncpack` (in fact we're going to have to use it on this repo soon enough) - but I'm sure there a few tools that solve the same issue out there!
> – @tknickman 

Would you be open to me raising a PR to set up syncpack in this repo? Would be a good opportunity for me to use it as a living example to create an onboarding tutorial at the same time. No pressure, just an idea 👍
